test/SILOptimizer/diagnose_unreachable.swift

```diff
@@ -470,3 +470,36 @@ extension Collection {
   }
 }
 
+// rdar://80415811 Incorrect unreachable code diagnostic for keypath to empty Enum
```

Comment:
I think it may be better to replace this comment by a more descriptive comment that explains the tests and the expectations.

Reply:
Also, I wonder if we should add a test that directly checks that the SIL location of the keypath_getter/setter is autogenerated. Can you check if there are SILGen tests that check SIL locations and if so, we can add a test there.

Reply:
Sure thing, added a test for autogenerated SIL locations.